lib/compiler/visitor.js

```diff
@@ -3,45 +3,35 @@
 // Simple AST node visitor builder.
 const visitor = {
   build(functions) {
-    function visit(node) {
-      return functions[node.type].apply(null, arguments);
+    function visit(node, ...args) {
+      return functions[node.type](node, ...args);
     }
 
     function visitNop() {
       // Do nothing.
     }
 
-    function visitExpression(node) {
-      const extraArgs = Array.prototype.slice.call(arguments, 1);
-
-      visit.apply(null, [node.expression].concat(extraArgs));
+    function visitExpression(node, ...args) {
+      return visit(node.expression, ...args);
     }
 
     function visitChildren(property) {
-      return function(node) {
-        const extraArgs = Array.prototype.slice.call(arguments, 1);
-
-        node[property].forEach(child => {
-          visit.apply(null, [child].concat(extraArgs));
-        });
+      return function(node, ...args) {
+        node[property].forEach(child => visit(child, ...args));
```

Comment:
I'm a little surprised this doesn't `return node[property].map()`.  Is that intentional?

Reply:
Yes. As David [explained in my first attempt](https://github.com/pegjs/pegjs/pull/393) to get it merged, he don't like to creation of an extra array that could not be used. I can understand that, and he arguments looks reasonable, I'm agree with them for that method. In general, I have already described my position by link, it has not changed, so I just refer to it

Reply:
OK, I'm fine with that, then.

Reply:
I still decided to add a comment to the code explaining why there is no `.map()`